src/main/scala/io/opentargets/etl/literature/Grounding.scala

```diff
@@ -152,27 +192,36 @@ object Grounding extends Serializable with LazyLogging {
     val scoreCN = "factor"
     val scoreC = col(scoreCN)
 
+    val selelectedCols = "pmid" :: "pmcid" :: "type" :: "label" :: "labelN" :: "keywordId" :: scoreCN :: Nil
+
+    logger.info("ground and take rank 1 from the mapped ones")
     val w = Window.partitionBy($"type", $"labelN").orderBy(scoreC.desc)
     val mappedLabel = labels
-      .select("type", "label", "labelN")
       .join(luts, Seq("type", "labelN"), "left_outer")
       .withColumn("isMapped", $"keywordId".isNotNull)
       .filter($"isMapped" === true)
+      .select(selelectedCols.map(col): _*)
+      .repartition(2048)
```

Comment:
Any reason for '2048'? Performance tuning?

Reply:
@JarrodBaker I am tryuing to improve the segments to process but I realised that is not the place to put it as it writes what it wants.